server/src/main/java/org/elasticsearch/bootstrap/ESPolicy.java

```diff
@@ -157,16 +158,17 @@ public boolean implies(ProtectionDomain domain, Permission permission) {
     }
 
     @SuppressForbidden(reason = "We get given an URL by the security infrastructure")
-    private boolean canAccessSecuredFile(URL location, FilePermission permission) {
-        if (location == null) {
+    private boolean canAccessSecuredFile(ProtectionDomain domain, FilePermission permission) {
+        if (domain == null) {
             return false;
         }
+        URL location = domain.getCodeSource().getLocation();
 
         // check the source
         Set<URL> accessibleSources = securedFiles.get(permission);
         if (accessibleSources != null) {
             // simple case - single-file referenced directly
-            return accessibleSources.contains(location);
+            return accessibleSources.contains(location) || system.implies(domain, new AllPermission());
         } else {
```

Comment:
I can't highlight the actual line, but there is another call to contains in the else here. Does that also need to have this additional check?